src/EmDeeCode.f90

```diff
@@ -358,7 +358,22 @@ subroutine add_item( excluded, i, j )
         integer :: start, end
         start = excluded%first(i)
         end = excluded%last(i)
-        if ((end < start).or.(j > excluded%item(end))) then
+! ORIGINAL INTENTION:
+!        if ((end < start) .or. (j > excluded%item(end)) &
+!          ) then
+!          excluded%item(end+2:n+1) = excluded%item(end+1:n)
+!          excluded%item(end+1) = j
+!        else
+!          do while (j > excluded%item(start))
+!            start = start + 1
+!          end do
+!          if (j == excluded%item(start)) return
+!          excluded%item(start+1:n+1) = excluded%item(start:n)
+!          excluded%item(start) = j
+!          start = start + 1
+!        end if
+! ALTERNATIVE IMPLEMENTATION
+        if ( (end < start) .or. jgteie(j,excluded,end) ) then
```

Comment:
May I suggest a cleaner solution? It is:

        if ((end >= start).and.(j <= excluded%item(end))) then
          do while (j > excluded%item(start))
            start = start + 1
          end do
          if (j == excluded%item(start)) return
          excluded%item(start+1:n+1) = excluded%item(start:n)
          excluded%item(start) = j
        else
          excluded%item(end+2:n+1) = excluded%item(end+1:n)
          excluded%item(end+1) = j
        end if